data-layers/data-layer-api-caller/src/main/java/com/flipkart/aesop/apicallerdatalayer/headers/ContextualHeaderProvider.java

```diff
@@ -0,0 +1,22 @@
+package com.flipkart.aesop.apicallerdatalayer.headers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by aman.gupta on 10/12/15.
+ */
+public class ContextualHeaderProvider extends HeaderCreater {
```

Comment:
Make the ContextualheaderProvider & StaticheaderProvider into interfaces.

Refactor the current implementation of thes header provides to something like string based header providers.